lib/constants.go

```diff
@@ -335,11 +345,12 @@ type MigrationHeight struct {
 }
 
 const (
-	DefaultMigration                     MigrationName = "DefaultMigration"
-	UnlimitedDerivedKeysMigration        MigrationName = "UnlimitedDerivedKeysMigration"
-	AssociationsAndAccessGroupsMigration MigrationName = "AssociationsAndAccessGroupsMigration"
-	BalanceModelMigration                MigrationName = "BalanceModelMigration"
-	ProofOfStakeNewTxnTypesMigration     MigrationName = "ProofOfStakeNewTxnTypesMigration"
+	DefaultMigration                       MigrationName = "DefaultMigration"
+	UnlimitedDerivedKeysMigration          MigrationName = "UnlimitedDerivedKeysMigration"
+	AssociationsAndAccessGroupsMigration   MigrationName = "AssociationsAndAccessGroupsMigration"
+	BalanceModelMigration                  MigrationName = "BalanceModelMigration"
+	ProofOfStake1StateSetupMigration       MigrationName = "ProofOfStake1StateSetupMigration"
+	ProofOfStake2ConsensusCutoverMigration MigrationName = "ProofOfStake2ConsensusCutoverMigration"
```

Comment:
Are any encoders actually changing  with this block height? If not, we don't need an encoder migration. Adding an encoder migration will cause the node to iterate over all the index to compute the post-migration state. It's a bit of an expensive operation so if we can only have one migration, that's preferable.

Reply:
I'll delete the ConsensusCutoverMigration for now. It's not used yet, but I think we will need it. But trivial to add back when we do. We use the StateSetupMigration for e.g. gating the new fields on a UtxoOperation.

Reply:
I guess my thinking is that if we KNOW we are going to be performing a migration later, we can perform it now without consequence. For example, if we want to start including a new field, but it isn't relevant until the 2nd block height, we can just encode its default value.